src/Command/Ssh/SshKeyCommandBase.php

```diff
@@ -31,15 +66,149 @@ public static function getIdeSshKeyLabel(IdeResponse $ide): string {
+   *   The filepath of the private SSH key.
+   * @param string $password
+   *
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function addSshKeyToAgent(string $filepath, string $password): void {
+    // We must use a separate script to mimic user input due to the limitations of the `ssh-add` command.
+    // @see https://www.linux.com/topic/networking/manage-ssh-key-file-passphrase/
+    $temp_filepath = $this->localMachineHelper->getFilesystem()->tempnam(sys_get_temp_dir(), 'acli');
+    $this->localMachineHelper->writeFile($temp_filepath, <<<'EOT'
+#!/usr/bin/env bash
+echo $SSH_PASS
+EOT
+    );
+    $this->localMachineHelper->getFilesystem()->chmod($temp_filepath, 0755);
+    $private_key_filepath = str_replace('.pub', '', $filepath);
+    $process = $this->localMachineHelper->executeFromCmd('SSH_PASS=' . $password . ' DISPLAY=1 SSH_ASKPASS=' . $temp_filepath . ' ssh-add ' . $private_key_filepath, NULL, NULL, FALSE);
+    $this->localMachineHelper->getFilesystem()->remove($temp_filepath);
+    if (!$process->isSuccessful()) {
+      throw new AcquiaCliException('Unable to add the SSH key to local SSH agent:' . $process->getOutput() . $process->getErrorOutput());
+    }
+  }
+
+  /**
+   * Polls the Cloud Platform until a successful SSH request is made to the dev
+   * environment.
+   *
+   * @param \Symfony\Component\Console\Output\OutputInterface $output
+   *
+   * @throws \Exception
+   */
+  protected function pollAcquiaCloudUntilSshSuccess(
+    OutputInterface $output
+  ): void {
+    // Create a loop to periodically poll the Cloud Platform.
+    $loop = Loop::get();
+    $spinner = LoopHelper::addSpinnerToLoop($loop, 'Waiting for the key to become available on the Cloud Platform', $output);
+
+    // Wait for SSH key to be available on a web.
+    $cloud_app_uuid = $this->getAnyAhApplication();
+    $environment = $this->getAnyAhEnvironment($cloud_app_uuid);
+
+    // Poll Cloud every 5 seconds.
+    $loop->addPeriodicTimer(5, function () use ($output, $loop, $environment, $spinner) {
+      try {
+        $process = $this->sshHelper->executeCommand($environment, ['ls'], FALSE);
+        if ($process->isSuccessful()) {
+          LoopHelper::finishSpinner($spinner);
+          $loop->stop();
+          $output->writeln("\n<info>Your SSH key is ready for use!</info>\n");
+        }
+        else {
+          $this->logger->debug($process->getOutput() . $process->getErrorOutput());
+        }
+      } catch (AcquiaCliException $exception) {
+        // Do nothing. Keep waiting and looping and logging.
+        $this->logger->debug($exception->getMessage());
+      }
+    });
+    LoopHelper::addTimeoutToLoop($loop, 15, $spinner);
+    $loop->run();
+  }
+
+  /**
+   * Get the first Cloud application available.
+   *
+   * @return false|mixed|string|null
+   * @throws \Exception
+   */
+  protected function getAnyAhApplication() {
+    if ($app_uuid = $this->determineCloudApplication(FALSE, FALSE)) {
```

Comment:
Not sure if this is the right decision. If they've already got an application linked to the current repo, we check the envs for that app for ssh connectivity. Otherwise, we just grab the first application they have access to and check that. We never ask them which app to test.

@anavarre @danepowell thoughts?

Reply:
I think we should force selecting/linking to a specific application to run this command.

A customer can have application access (i.e. UI / API access) without having SSH access. For this to work without user input, we'd have to separately query the permissions API to figure out which applications allow SSH access. Even then, it's not ideal since different servers might install the SSH keys at different times. Just because one works doesn't mean the others will.

Reply:
Ah.

Reply:
Trouble with this is that the user may be confused about why we're asking them to select an application.

Reply:
Let's just be transparent, we need them to link the app so we can verify that the SSH key was installed correctly. We can make it optional: if they don't want to link the app, we won't verify the installation.